### PR TITLE
fix(dashboard): Display boolean values in CustomCheckEditor

### PR DESCRIPTION
**Pull request by robertkasza**
### Checklist

- [x] No breaking changes
- [x] Tests pass
- [x] New features have new tests
- [ ] Documentation is updated (if applicable)
- [x] Title of the PR is in the correct format (see below)

### **PR Type**
Bug fix

___

### **Description**
- Fixes the variable-combobox trigger label in `CustomCheckEditor` so non-string `comboboxValue`s (e.g. boolean `true`/`false`) render as readable text instead of falling back to the placeholder or breaking the trigger.
- Introduces a small `getDisplayValue` helper that returns strings as-is, treats `null`/`undefined` as empty so the `'Select variable...'` placeholder still wins, and `String()`-coerces other primitives.
- Adds a parametrized vitest case covering boolean, string, `null`, and `undefined` values to lock in the trigger label rendering.

___

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody>
<tr><td><strong>Core logic</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>ConditionValue.tsx</strong><dd><code>Add getDisplayValue helper to safely stringify non-string combobox values for the trigger label.</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4228/files#diff-810657158af1493e2b841f61a1dfa02c99077838c527e860607fd0c5ca311f08">+13/-1</a></td>
</tr>
</table></details></td></tr>
<tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>ConditionValue.test.tsx</strong><dd><code>Add parametrized trigger-label test covering boolean, string, null, and undefined values.</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4228/files#diff-19fa3f88d3784d5a287e92b6b48ca0e9542916589235e0a00372d75f79abd5a4">+25/-0</a></td>
</tr>
</table></details></td></tr>
</tbody></table>

</details>

___

### CODE REVIEW

File: dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx [hunk 1]
```diff
@@ -236,8 +278,14 @@ function ConditionValue({
   const selectedVariable = availableHasuraPermissionVariables.find(
     (variable) => variable.value === comboboxValue,
   );
+  const displayValue =
+    typeof comboboxValue === 'string'
+      ? comboboxValue
+      : comboboxValue == null
+        ? ''
+        : String(comboboxValue);
```
Comment (dbm03) [inline dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx]: nit: nested ternaries are usually advised against

File: dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx [hunk 2]
```diff
@@ -236,8 +278,14 @@ function ConditionValue({
...
   const comboboxLabel =
-    selectedVariable?.label || comboboxValue || 'Select variable...';
+    selectedVariable?.label || displayValue || 'Select variable...';
```
Comment (dbm03) [inline dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx]: Before this PR's changes, we used to show in the UI whether the selected value was something different to `true|false`, e.g. "X-Hasura-User-Id".

```json
{
  "booleano": {
    "_eq": "X-Hasura-User-Id"
  }
}
```

Before:
Image

After (this PR's branch):
Image
The value seems empty but it's actually "X-Hasura-User-Id" if you go to the JSON tab

I believe it'd be better to show the actual selected value (even if it's not selectable in the `<Select>` component)

Comment (robertkasza) [reply dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx]: I think you are right, but that check would never works. But we need to support env variables.

Comment (robertkasza) [reply dashboard/src/features/orgs/projects/database/dataGrid/components/CustomCheckEditor/ConditionValue.tsx]: Good catch by the way. I will revert the changes and fix the issue that i noticed that true is not displayed.
image

**Review by dbm03 — APPROVED** [review]
